components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/impl/ApisApiServiceImpl.java

```diff
@@ -734,7 +734,7 @@ public Response apisApiIdGet(String apiId, String accept, String ifNoneMatch, St
         Map<String, Integer> permissionMap = api.getPermissionMap();
 
         //TODO: Remove the check for admin after IS adds an ID to admin user
-        if (!permissionMap.isEmpty() && !"admin".equals(loggedInUserName)) {
+        if (permissionMap != null && !permissionMap.isEmpty() && !"admin".equals(loggedInUserName)) {
```

Comment:
Although this is not related to your fix, isn't that wrong to use hardcoded 'admin' username here?

Reply:
Thanks for the review. This is fixed in the latest code.